File: AtCoder/ABC/211/python/C-chokudai.py
MOD = 10**9 + 7
S = input()

C = "chokudai"
c2i = dict(zip(C, range(len(C))))
d = {k: 0 for k in C}

for c in S:
    if c in d:
        prv = C[c2i[c] - 1]
        if c == 'c':
            d[c] += 1
        else:
            d[c] = (d[c] + d[prv]) % MOD

print(d['i'])

File: AtCoder/ABC/211/cpp/C-chokudai.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

const ll MOD = 1e9 + 7;
const string T = "chokudai";
string S;

int main() {
  map<char, ll> c2i;
  for (ll i = 0; i < T.size(); ++i) {
    c2i[T[i]] = i;
  }
  cin >> S;
  map<char, ll> C;
  for (auto c : S) {
    if (c2i.find(c) == c2i.end()) continue;
    if (c == 'c') {
      ++C[c];
      continue;
    }
    char prv = T[c2i[c] - 1];
    C[c] = (C[c] + C[prv]) % MOD;
  }
  printf("%lld\n", C['i']);
}